components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -365,9 +382,12 @@ export const useSwap = () => {
     [quoteOptions, toToken]
   )
 
+  const fromAssetBalance = fromToken && getAssetBalance(fromToken, fromAccount, tokenBalancesRegistry)
+  const nativeAssetBalance = nativeAsset && getAssetBalance(nativeAsset, fromAccount, tokenBalancesRegistry)
```

Comment:
perf: since `getAssetBalance` creates an object, we should memoize these to prevent re-renders since